src/PayU/Alu/Client.php

```diff
@@ -177,20 +62,26 @@ public function pay(Request $request, HTTPClient $httpClient = null, HashService
             $httpClient = new HTTPClient();
         }
 
-        $requestHash = $hashService->makeRequestHash($request);
-        $request->setOrderHash($requestHash);
-
-        $requestParams = $request->getRequestParams();
-        $responseXML = $httpClient->post($this->getAluUrl(), $requestParams);
         try {
-            $xmlObject = new SimpleXMLElement($responseXML);
-        } catch (\Exception $e) {
-            throw new ClientException($e->getMessage(), $e->getCode());
+            $gateway = $this->authorizationFactory->create(
+                $request->getPaymentsApiVersion(),
+                $httpClient,
+                $hashService
+            );
+        } catch (AuthorizationFactoryException $exception) {
+            throw new ClientException($exception->getMessage());
+        }
+
+        if ($this->customUrl !== null) {
```

Comment:
customUrl  should be removed because it is not used anymore

Reply:
Done - I've marked the method as deprecated.